src/snowflake/connector/time_util.py

```diff
@@ -47,17 +50,72 @@ def get_time_millis() -> int:
     return int(time.time() * 1000)
 
 
-class DecorrelateJitterBackoff:
-    # Decorrelate Jitter backoff
-    # https://www.awsarchitectureblog.com/2015/03/backoff.html
-    def __init__(self, base: int, cap: int) -> None:
+class BackoffPolicy(ABC):
```

Comment:
I still think that we should at least explore using Generators instead of classes. There are so few `super` calls that I'm not sure if using OOP here is warranted

Reply:
I've been learning a lot about generators and coroutines while learning about asyncio. I can definitely look into working in generators then (right now) if that sounds good?

Reply:
You cannot change the public API once this gets released!